src/configuration/applier/state.cc

```diff
@@ -642,21 +672,49 @@ void applier::state::_check_hostescalations() const {
 
     for (auto const& escalation : hst->get_escalations()) {
       bool found = false;
-      for (auto const& e : engine::serviceescalation::serviceescalations) {
+      for (auto const& e : engine::hostescalation::hostescalations) {
         if (e.second.get() == escalation) {
           found = true;
           break;
         }
       }
       if (!found) {
         logger(log_config_error, basic)
-          << "Error on hostescalation !!! The host "
-          << hst->get_name()
-          << " contains a non existing host escalation";
+            << "Error on hostescalation !!! The host " << hst->get_name()
+            << " contains a non existing host escalation";
         throw engine_error() << "This is a bug";
       }
     }
   }
+
+  for (auto const& e : engine::hostescalation::hostescalations) {
+    engine::hostescalation const* he{e.second.get()};
+    bool found = false;
+
+    for (auto const& p : engine::host::hosts) {
```

Comment:
perhaps we could do a engine::host::hosts.find(he->notifier_ptr->get_hostname) . But I am not sure, it seems quicker but more dangerous ...